### PR TITLE
fix: --title --kind --custom-property parameters

### PR DESCRIPTION
**Pull request by blackchoey**
Use latest aaz-dev to regenerate CLI code to fix following parameter issues:
1. `--title` and `--kind` should be required
2. `--custom-properties` should be free-form object

### CODE REVIEW

File: src/apic-extension/azext_apic_extension/aaz/latest/apic/api/definition/_export_specification.py [hunk 1]
```diff
@@ -124,24 +123,10 @@ class ApiDefinitionsExportSpecification(AAZHttpOperation):
         def __call__(self, *args, **kwargs):
             request = self.make_request()
             session = self.client.send_request(request=request, stream=False, **kwargs)
-            if session.http_response.status_code in [202]:
-                return self.client.build_lro_polling(
-                    self.ctx.args.no_wait,
-                    session,
-                    self.on_200,
-                    self.on_error,
-                    lro_options={"final-state-via": "location"},
```
Comment (arpishahmsft) [inline src/apic-extension/azext_apic_extension/aaz/latest/apic/api/definition/_export_specification.py:133]: this is an async operation, should not be removed. Please check spec for async polling properties on location

File: src/apic-extension/azext_apic_extension/aaz/latest/apic/api/definition/_export_specification.py [hunk 2]
```diff
@@ -124,24 +123,10 @@ class ApiDefinitionsExportSpecification(AAZHttpOperation):
...
-                    path_format_arguments=self.url_parameters,
-                )
             if session.http_response.status_code in [200]:
-                return self.client.build_lro_polling(
```
Comment (arpishahmsft) [inline src/apic-extension/azext_apic_extension/aaz/latest/apic/api/definition/_export_specification.py:137]: this is an async operation, should not be removed. Please check spec for async polling properties on location

File: src/apic-extension/azext_apic_extension/aaz/latest/apic/api/definition/_export_specification.py [hunk 3]
```diff
@@ -28,11 +28,10 @@ class ExportSpecification(AAZCommand):
         ]
     }
 
-    AZ_SUPPORT_NO_WAIT = True
-
     def _handler(self, command_args):
         super()._handler(command_args)
-        return self.build_lro_poller(self._execute_operations, self._output)
```
[the remark this answers is not in the document]
Comment (blackchoey) [reply src/apic-extension/azext_apic_extension/aaz/latest/apic/api/definition/_export_specification.py:35]: The spec is the same. I'm using latest aaz-dev, which generates different code. Same for other similar changes.
I can double confirm whether this is expected with Azure CLI team.

[the remark this answers is not in the document]
Comment (blackchoey) [reply src/apic-extension/azext_apic_extension/aaz/latest/apic/api/definition/_export_specification.py]: It's an issue with the spec, the 2024-03-01 version misses the long running operation field. Reverted these changes.

**Pull request merged**